pkg/controllermanager/controller/controllerregistration/controllerregistration_seed_control.go

```diff
@@ -42,129 +39,92 @@ import (
-}
-
-// RegistrationSeedControlInterface implements the control logic for updating Seeds. It is implemented as an interface to allow
-// for extensions that provide different semantics. Currently, there is only one implementation.
-type RegistrationSeedControlInterface interface {
-	Reconcile(*gardencorev1beta1.Seed) error
-}
-
-// NewDefaultControllerRegistrationSeedControl returns a new instance of the default implementation ControlInterface that
-// implements the documented semantics for Seeds. You should use an instance returned from NewDefaultControllerRegistrationSeedControl()
-// for any scenario other than testing.
-func NewDefaultControllerRegistrationSeedControl(
-	clientMap clientmap.ClientMap,
-	secretLister corev1listers.SecretLister,
-	backupBucketLister gardencorelisters.BackupBucketLister,
-	controllerInstallationLister gardencorelisters.ControllerInstallationLister,
-	controllerRegistrationLister gardencorelisters.ControllerRegistrationLister,
-	seedLister gardencorelisters.SeedLister,
-) RegistrationSeedControlInterface {
-	return &defaultControllerRegistrationSeedControl{clientMap, secretLister, backupBucketLister, controllerInstallationLister, controllerRegistrationLister, seedLister}
 }
 
-type defaultControllerRegistrationSeedControl struct {
-	clientMap                    clientmap.ClientMap
-	secretLister                 corev1listers.SecretLister
-	backupBucketLister           gardencorelisters.BackupBucketLister
-	controllerInstallationLister gardencorelisters.ControllerInstallationLister
-	controllerRegistrationLister gardencorelisters.ControllerRegistrationLister
-	seedLister                   gardencorelisters.SeedLister
+type controllerRegistrationSeedReconciler struct {
+	logger       logrus.FieldLogger
+	gardenClient kubernetes.Interface
 }
 
-// Reconcile reconciles the ControllerInstallations for given Seed object. It computes the desired list of
-// ControllerInstallations, deploys them, and deletes all other potentially existing installation objects.
-func (c *defaultControllerRegistrationSeedControl) Reconcile(obj *gardencorev1beta1.Seed) error {
-	var (
-		ctx    = context.TODO()
-		seed   = obj.DeepCopy()
-		logger = logger.NewFieldLogger(logger.Logger, "controllerregistration-seed", seed.Name)
-	)
+func (r *controllerRegistrationSeedReconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	seed := &gardencorev1beta1.Seed{}
+	if err := r.gardenClient.Client().Get(ctx, request.NamespacedName, seed); err != nil {
+		if apierrors.IsNotFound(err) {
+			r.logger.Infof("Object %q is gone, stop reconciling: %v", request.Name, err)
+			return reconcile.Result{}, nil
+		}
+		r.logger.Infof("Unable to retrieve object %q from store: %v", request.Name, err)
+		return reconcile.Result{}, err
+	}
 
+	logger := logger.NewFieldLogger(r.logger, "controllerregistration-seed", seed.Name)
 	logger.Infof("[CONTROLLERINSTALLATION SEED] Reconciling %s", seed.Name)
```

Comment:
Nit: you could omit the seed name when logging as the `logger` is now a field one.

```suggestion
	logger.Info("[CONTROLLERINSTALLATION SEED] Reconciling")
```